src/twisted/runner/procmon.py

```diff
@@ -5,11 +5,50 @@
 """
 Support for starting, monitoring, and restarting child process.
 """
-from twisted.python import log
+import attr
+import incremental
+
+from twisted.python import log, deprecate
 from twisted.internet import error, protocol, reactor as _reactor
 from twisted.application import service
 from twisted.protocols import basic
 
+@attr.s(frozen=True)
+class _Process(object):
+    """
+    The parameters of a process to be restarted.
+
+    @ivar args: command-line arguments (including name of command as first one)
+    @type args: C{list}
+
+    @ivar uid: user-id to run process as, or None
+    @type uid: C{int}
+
+    @ivar gid: group-id to run process as, or None
+    @type gid: C{int}
+
+    @ivar env: environment for process
+    @type env: C{dict}
+
+    @ivar cwd: initial working directory for process
+    @type cwd: C{str}
+    """
+
+    args = attr.ib()
+    uid = attr.ib(default=None)
+    gid = attr.ib(default=None)
+    env = attr.ib(default=attr.Factory(dict))
+    cwd = attr.ib(default=None)
+
+    @deprecate.deprecated(incremental.Version("Twisted", "NEXT", 0, 0))
+    def toTuple(self):
+        """
+        Convert to tuple
```

Comment:
just a note the other devs:

I am not sure if this docstrings adds any extra info on top of `def toTuple(self):` and if it makes sense to use it

So in the current form it's just useless documentation.

Maybe use this oportunity to add info about what to use instead of this deprecated code